client/src/java/com/zimbra/client/ZSharedFolder.java

```diff
@@ -38,6 +40,18 @@ public String getTargetId() {
         return targetId;
     }
 
+    public ItemIdentifier getTargetItemIdentifier() {
+        ItemIdentifier fId;
+        try {
+            fId = new ItemIdentifier(targetId, null);
+            return fId;
+        } catch (ServiceException e) {
+            ZimbraLog.mailbox.debug("Problem understanding folderId '%s' - assume hidden", targetId, e);
+            throw new RuntimeException(
```

Comment:
Hmm.  Strange, I thought I was doing this to avoid lots of error handling for something that should never fail but the only caller already throws ServiceException.  I'll update and push a new change.